lib/index.js

```diff
@@ -0,0 +1,251 @@
+      }
+
+      return Object.assign({}, locals, {
+        postcodeLabel: conditionalTranslate(`fields.${addressKey}-postcode.label`, req.translate) || defaults.POSTCODE_LABEL,
+        route: this.options.route,
+        editLink,
+        cantFind,
+        postcodeError,
+        postcode,
+        section
+      });
+    }
+
+    // eslint-disable-next-line consistent-return
+    process(req, res, callback) {
+      if (req.query.step !== 'postcode') {
+        return super.process(req, res, callback);
+      }
+      const postcode = req.form.values[`${addressKey}-postcode`];
+      const previousPostcode = req.sessionModel.get(`${addressKey}-postcode`);
+      if (!postcode
+        || previousPostcode && previousPostcode === postcode) {
+        return callback();
+      }
+
+      this.model.fetch(postcode)
+        .then(data => {
+          if (data.length) {
+            req.sessionModel.set(`${addressKey}-addresses`, data);
+          } else {
+            req.sessionModel.unset(`${addressKey}-addresses`);
+            req.sessionModel.set(`${addressKey}-postcodeApiMeta`, {
+              messageKey: 'not-found'
+            });
+          }
+          return callback();
+        })
+        .catch(err => {
+          req.form.options.apiError = true;
+          if (err.status === 501) {
+            req.sessionModel.unset([
+              `${addressKey}-postcodeApiMeta`,
+              `${addressKey}-addresses`
+            ]);
+            return callback();
+          }
+          req.sessionModel.set(`${addressKey}-postcodeApiMeta`, {
+            messageKey: 'cant-connect'
+          });
+          // eslint-disable-next-line no-console
+          console.error('Postcode lookup error: ',
+            `Code: ${err.status}; Detail: ${err.detail}`);
+          return callback();
+        });
+    }
+
+    saveValues(req, res, callback) {
+      if (req.query.step === 'lookup') {
+        const addressLines = req.form.values[`${addressKey}-select`].split(', ').join('\n');
+        req.sessionModel.set(addressKey, addressLines);
+      }
+      super.saveValues(req, res, callback);
+    }
+
+    // eslint-disable-next-line consistent-return
+    validate(req, res, callback) {
+      if (req.query.step === 'postcode' &&
+        this.model.validate &&
+        !req.form.options.apiError
+      ) {
+        const key = `${addressKey}-postcode`;
+        const postcode = encodeURIComponent(req.form.values[key]);
+        this.model.validate(postcode)
+          .then(json => {
+            return callback();
+          })
+          .catch(err => {
+            if (err.status === 418) {
+              err = {
+                [key]: new this.ValidationError(key, {
+                  key,
+                  type: err.type,
+                  redirect: undefined
+                }, req, res)
+              };
+            } else if (err.status === 403) {
+              return callback();
```

Comment:
We'll want to call `super.validate` here in case other behaviours have been applied.